src/provider/apiresponse.h

```diff
@@ -0,0 +1,54 @@
+#ifndef ERD_PROXY_API_RESPONSE_H
+#define ERD_PROXY_API_RESPONSE_H
+
+#include "json/json.hpp"
+#include "../utils/errors.h"
+
+class ErdGenericApiResponse
+{
+public:
+    explicit ErdGenericApiResponse(std::string rawData)
+    {
+        try
+        {
+            m_response = nlohmann::json::parse(rawData);
+        }
+        catch (...)
+        {
+            throw std::invalid_argument(ERROR_MSG_JSON_SERIALIZED + rawData);
+        }
+    }
+
+    std::string getError() const
+    {
+        std::string ret;
+
+        if (m_response.contains("error"))
+            if(m_response["error"] != "")
+                ret = m_response["error"];
+
+       return ret;
+    }
+
+    std::string getCode() const
+    {
+        if (!m_response.contains("code"))
```

Comment:
Here, and below - we can define an extra function called `requireAttribute()` and remove bits of duplication. Not mandatory.

Reply:
Added it !